doppler/resource_project.go

```diff
@@ -0,0 +1,101 @@
+package doppler
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceProject() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceProjectCreate,
+		ReadContext:   resourceProjectRead,
+		UpdateContext: resourceProjectUpdate,
+		DeleteContext: resourceProjectDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the Doppler project",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "The description of the Doppler project",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceProjectCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(APIClient)
+
+	var diags diag.Diagnostics
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+
+	project, err := client.CreateProject(ctx, name, description)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(project.Id)
+
+	return diags
+}
+
+func resourceProjectUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(APIClient)
+
+	var diags diag.Diagnostics
+	currentName := d.Id()
```

Comment:
Is this the reason why we have to recreate everything when we wanna change the name? Could we use the slug instead?

Reply:
Oh, I see what you mean now! When you change a project's name, that trickles into the configs/environments and Terraform forces recreation when it wouldn't actually need to. I totally didn't realize that when I put up the demo.

Unfortunately, the project slug is the same as the project's name so Terraform won't be able to distinguish between a rename or a recreation of the whole project. This is definitely a bummer because you'll lose your config logs in this process when you didn't really need to.

I think this situation is rare enough to accept this tradeoff but I agree, it isn't ideal.

Reply:
Ooh I see, because the environments are using the project name as the identifier. 

I agree that renaming a project is probably not something that happens super often, so probably okay